### PR TITLE
Add a new trust bundle for custom SSL certificates

### PR DESCRIPTION
**Pull request by aledbf**
## Description

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at 3b41472</samp>

Add support for custom SSL certificates in cert-manager. Use a new variable and a config map to provide custom CA certificates to cert-manager pods.

</details>

## Related Issue(s)

Fixes #

## How to test
- Check the configmap `gitpod-customer-certificate-bundle` exists
- Check the file `/etc/ssl/certs/ca-certificates.crt` in a workspace is mounted volume from the `ca-certificates` configmap
- Configure the installer with a custom CA certificate and check the `/etc/ssl/certs/ca-certificates.crt` contains the certificate

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - aledbf-cus2745037e55</li>
	<li><b>🔗 URL</b> - <a href="https://aledbf-cus2745037e55.preview.gitpod-dev.com/workspaces" target="_blank">aledbf-cus2745037e55.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - aledbf-custom-cert-trust-gha.14838</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/ws-manager-mk2/controllers/create.go [hunk 1]
```diff
@@ -315,6 +315,14 @@ func createDefiniteWorkspacePod(sctx *startWorkspaceContext) (*corev1.Pod, error
 				},
 			},
 		},
+		{
+			Name: "ca-certificates",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{Name: "gitpod-customer-certificate-bundle"},
```
[the remark this answers is not in the document]
Comment (aledbf) [reply components/ws-manager-mk2/controllers/create.go:322]: done

**Review by nandajavarma — APPROVED** [review]

**Pull request merged**